File: app/src/main/java/com/hamburgcodingschool/weather/MainActivity.java
package com.hamburgcodingschool.weather;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity implements WeatherView {

    private TextView weatherTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        weatherTextView = findViewById(R.id.hello);

        WeatherRepository repository = new WeatherRepository(getString(R.string.OpenWeatherMapKey));
        WeatherPresenter presenter = new WeatherPresenter(this, repository);
        repository.setCallback(presenter);
        presenter.getWeather("Hamburg");

        Log.d("MainActivity", "We created a weather activity");
    }

    @Override
    public void setWeather(String weather) {
        weatherTextView.setText(weather);
    }
}

File: app/src/main/java/com/hamburgcodingschool/weather/MainActivity.kt
package com.hamburgcodingschool.weather

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.hello

class MainActivity : AppCompatActivity(), WeatherView {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val repository = WeatherRepository(getString(R.string.OpenWeatherMapKey))
        val presenter = WeatherPresenter(this, repository)
        repository.setCallback(presenter)
        presenter.getWeather("Hamburg")

        Log.d("MainActivity", "We created a weather activity")
    }

    override fun setWeather(weather: String) {
        hello.text = weather
    }
}
